akka-persistence/src/main/scala/akka/persistence/keyvaluestore/scaladsl/KeyValueUpdateStore.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2009-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.keyvaluestore.scaladsl
+
+import scala.concurrent.Future
+
+import akka.Done
+
+/**
+ * API for updating key-value objects.
+ *
+ * For Java API see [[akka.persistence.keyvaluestore.javadsl.KeyValueUpdateStore]].
+ */
+trait KeyValueUpdateStore[A] extends KeyValueStore[A] {
```

Comment:
When implementing a `KeyValueStore` this (and corresponding javadsl) is what needs to be implemented. I separated in read KeyValueStore and write KeyValueUpdateStore, because read KeyValueStore can also be useful on the query side, see `KeyValueQueryStore`.

Reply:
@debasishg What I would suggest as next step would be an in-memory in-memory implementation of this and start adding tests using that. Event sourced tests for inspiration: https://github.com/akka/akka/tree/master/akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl

Reply:
Thanks @patriknw .. I will start taking a look at `EventSourcedBehavior` and the related implementations and start writing the in memory implementation.

Reply:
@patriknw This is from the code ..

```scala
/**
  * @param seqNr sequence number for optimistic locking. starts at 1.
  */
  def upsertObject(persistenceId: String, seqNr: Long, value: A, tag: String): Future[Done]
```
* For an in-memory implementation based on a `Map`, the `value` to be persisted can be anything - right ? So the operations are just same as manipulating a `Map`. Or do we have to consider messages in the form `InmemMessages` as in https://github.com/akka/akka/blob/master/akka-persistence/src/main/scala/akka/persistence/journal/inmem/InmemJournal.scala#L140 ?
* How about `seqNr` and `tag` ?  Do they have any meaning with an in-memory implementation ? Do we consider tagged messages in in-memory impl ?